src/vcy/vcy_parser.mly

```diff
@@ -254,6 +255,9 @@ stmt:
   | ASSERT e=exp SEMI { loc $startpos $endpos @@ Assert e }
   | ASSUME e=exp SEMI { loc $startpos $endpos @@ Assume e }
   | HAVOC i=IDENT SEMI { loc $startpos $endpos @@ Havoc i }
+  | variant=commute_variant phi=commute_condition
+    LBRACE PRE COLON pre=exp blocks=nonempty_list(block) POST COLON post=exp RBRACE
```

Comment:
Merge this rule with the above rule while making the parsing of the pre/post an `Option`.